Library/Homebrew/resource.rb

```diff
@@ -145,7 +145,10 @@ def url(val = nil, specs = {})
   end
 
   def version(val = nil)
-    @version ||= detect_version(val)
+    @version ||= begin
+      version = detect_version(val)
+      version.is_a?(Version::NullVersion) ? nil : version 
```

Comment:
This probably isn't the right fix, but I didn't ferret out where the `nil` check is happening. This fixed an issue where resources w/o parseable versions weren't inheriting the version of their parent since they now had non-nil values.